documentation/src/main/doc/modules/kafka/examples/inbound/KafkaRebalancedConsumerRebalanceListener.java

```diff
@@ -0,0 +1,71 @@
+package inbound;
+
+import io.smallrye.mutiny.Uni;
+import io.smallrye.reactive.messaging.kafka.KafkaConsumerRebalanceListener;
+import io.vertx.kafka.client.common.TopicPartition;
+import io.vertx.mutiny.kafka.client.consumer.KafkaConsumer;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Named;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+@Named("rebalanced-example.rebalancer")
+public class KafkaRebalancedConsumerRebalanceListener implements KafkaConsumerRebalanceListener {
+
+    private static final Logger LOGGER = Logger.getLogger(KafkaRebalancedConsumerRebalanceListener.class.getName());
+
+    /**
+     * When receiving a list of partitions will search for the earliest offset within 10 minutes
+     * and seek the consumer to it. These operations are asynchronous so the inbound connector's
+     * consumer WILL be paused until they are complete.
+     *
+     * @param consumer        underlying consumer
+     * @param topicPartitions set of assigned topic partitions
+     * @return An observable
+     */
+    @Override
+    public Uni<Void> onPartitionsAssigned(KafkaConsumer<?, ?> consumer, Set<TopicPartition> topicPartitions) {
+        // we must pause the consumer otherwise the inbound connector will continue to receive messages
+        // while we seek to the correct offset.
+        consumer.pause();
```

Comment:
Question: should it be done by the connector, or by the implementation of the`KafkaConsumerRebalanceListener`?

Reply:
I thought about this. The listener isn't always used to seek to separate offset, so it's really up to the user.

Reply:
On further thought. It does make sense to do it. By the kafka documentation it is expected that the operations in the listener are completed before receiving events. If the user wishes to override they can resume from withing their implementation.

Reply:
Agreed.

Reply:
So it needs to be changed in this example and explained in the doc.